src/lib/converter/utils/types.ts

```diff
@@ -0,0 +1,45 @@
+import * as ts from 'typescript';
+
+/**
+ * Returns the type parameters of a given type.
+ * @param type The type whos type parameters are wanted.
+ * @returns The type parameters of the type. An empty array if the type has no type parameters.
+ */
+export function getTypeParametersOfType(type: ts.Type): ts.TypeParameterDeclaration[] {
+    for (const declaration of type.symbol.declarations) {
+        if ((ts.isClassDeclaration(declaration) || ts.isInterfaceDeclaration(declaration)) &&
+             declaration.typeParameters) {
+            return declaration.typeParameters.map(tp => tp);
```

Comment:
Why the map? To remove `readonly` from the array type? I'd rather just annotate this as returning a readonly array.

Reply:
I did it, because I couldn't use the `NodeArray` type as input to `Array.concat` later on. I find these TS types a big confusing. But `ReadonlyArray` is the better solution.